playwright/tests/a11y.spec.js

```diff
@@ -1,66 +1,145 @@
+  'http://www.va.gov/burials-memorials/memorial-items/presidential-memorial-certificates/request-certificate-form-40-0247/',
+  'http://www.va.gov/my-health/order-hearing-aid-or-cpap-supplies-form/',
+  'http://www.va.gov/education/other-va-education-benefits/veteran-rapid-retraining-assistance/apply-for-vrrap-form-22-1990s/',
+  'http://www.va.gov/track-claims/',
+  'http://www.va.gov/burials-and-memorials-v2/application/530/',
+  'http://www.va.gov/profile/',
+  'http://www.va.gov/my-va/',
+  'http://www.va.gov/disability/view-disability-rating/rating/',
+  'http://www.va.gov/view-change-dependents/view/',
+  'http://www.va.gov/supporting-forms-for-claims/substitute-claimant-form-21P-0847/',
+  'http://www.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/',
+  'http://www.va.gov/careers-employment/education-and-career-counseling/apply-career-guidance-form-25-8832/',
+]
 
-test.describe('Accessibility Tests', async () => {
+test.describe(`Accessibility Site Scan`, async () => {
   test.setTimeout(4200000)
 
-  const pages = await getSitemapLocations(
-    process.env.BASE_URL || 'http://127.0.0.1:8001'
-  )
+  test(`The sitemap should contain accessible pages`, async ({
+    page,
+    browserName,
+    // makeAxeBuilder,
+  }, testInfo) => {
+    const viewportSize = page.viewportSize()
+    let scanResults = []
+    let failedPages = []
+    let pages = await getSitemapLocations(baseUrl)
 
-  let pageSegments
+    if (segmentNumber !== 0) {
+      pages = splitArray(pages, totalSegments)[segmentNumber - 1]
+    }
 
-  test.beforeAll(async () => {
-    pageSegments = await splitPagesIntoSegments(pages, 5)
-  })
+    console.log('number of pages in segment', pages.length)
+    console.log('browser name:', browserName)
+    console.log('viewport size:', viewportSize)
+
+    for (const pageUrl of pages) {
+      if (excludedPages.includes(pageUrl)) {
+        console.log('skipping scan of excluded page:', pageUrl)
+        continue
+      }
+
+      try {
+        console.log('testing page:', pageUrl)
+        await page.goto(pageUrl)
+
+        // @todo The shared "makeAxeBuilder" never reports errors for whatever reason so not using for now.
+        // const accessibilityScanResults = await makeAxeBuilder({ page }).analyze()
+
+        const accessibilityScanResults = await new AxeBuilder({ page })
+          .withTags(['section508', 'wcag2a', 'wcag2aa', 'wcag21a', 'wcag21aa'])
+          .exclude('iframe')
+          // Exclude header and footer since they are shared on every page.
+          .exclude('header')
+          .exclude('footer')
+          .analyze()
+
+        console.log('page violations:', accessibilityScanResults.violations)
+        if (accessibilityScanResults.violations.length > 0) {
+          scanResults.push({
+            url: pageUrl,
+            browserName,
+            viewportSize,
+            violations: accessibilityScanResults.violations,
```

Comment:
i think this object could be slightly more helpful / trimmed down some, with something like:

```js
scanResults.push({
             url: pageUrl,
             browserName,
             viewportSize,
             violations: accessibilityScanResults.violations.map((v) => {
                 id: v.id,
                 description: v.description,
                 impact: v.impact,
                 nodes: `${v.nodes.map((node) => node.html).join(', ')}`
             })
})
```

Reply:
What does the accessibility person reviewing the report want to see? I'm all for making it more helpful for them to review.

Reply:
the content-build reports have expired, so can't see the actual logs... but the file itself has this:

https://github.com/department-of-veterans-affairs/content-build/blob/main/src/platform/testing/e2e/cypress/support/commands/axeCheck.js#L22

so the violations are being trimmed somewhat. The example I shared above was basically modified from the original output of this scan prior to your updates:

```js
    if (accessibilityScanResults.violations.length > 0) {
      accessibilityScanResults.violations.forEach((violation) => {
        console.log(`URL: ${pageUrl}`)
        console.log(`Violation: ${violation.id}`)
        console.log(`Description: ${violation.description}`)
        console.log(`Impact: ${violation.impact}`)
        console.log(
          `Nodes: ${violation.nodes.map((node) => node.html).join(', ')}`
        )
      })

      a11yFailures.push({
        url: pageUrl,
        violations: accessibilityScanResults.violations,
      })
```

I think there's a happy medium in there somewhere... the default violation output includes a lot of stuff that I think isn't particularly relevant or helpful.

Reply:
Updated this part of the code to have a tidier report basically in the same shape.